GameServerLib/Content/Navigation/NavigationGrid.cs

```diff
@@ -803,15 +803,37 @@ public float GetHeightAtLocation(float x, float y)
             float dx = distx / greatestdist;
             float dy = disty / greatestdist;
             int i;
+            bool prevPosHadBush = HasFlag(origin, NavigationGridCellFlags.HAS_GRASS);
+
             for (i = 0; i < il; i++)
             {
-                //TODO: Implement bush logic (preferably near here)
                 //TODO: Implement methods for maps whose NavGrids don't use SEE_THROUGH flags for buildings
                 if ((checkWalkable && !IsWalkable(x1, y1)) || (checkVisible && !IsVisible(x1, y1)))
                 {
                     break;
                 }
 
+                bool isGrass = HasFlag(new Vector2(x1, y1), NavigationGridCellFlags.HAS_GRASS);
+                // If you are outside of a bush
+                if (checkVisible && !prevPosHadBush)
+                {
+                    if (isGrass)
+                    {
+                        break;
+                    }
+                }
+
+                // If you are in a different bush
+                if (checkVisible && prevPosHadBush && HasFlag(destination, NavigationGridCellFlags.HAS_GRASS))
```

Comment:
`HasFlag(destination, NavigationGridCellFlags.HAS_GRASS)` can also be computed outside of the loop I think, as the destination doesn't change.

Reply:
I could swear i had made that change o_0
Well here it is again(?)